ext/authx/Civi/Authx/Authenticator.php

```diff
@@ -64,6 +63,13 @@ public function on_civi_invoke_auth(GenericHookEvent $e) {
         _authx_redact(['_authx']);
       }
     }
+    // Accept legacy auth (?key=...&api_key=...) for 'civicrm/ajax/rest' and 'civicrm/ajax/api4/*'.
+    // The use of `?key=` could clash on some endpoints. Only accept on a small list of endpoints that are compatible with it.
+    if (count($e->args) > 2 && $e->args[1] === 'ajax' && in_array($e->args[2], ['rest', 'api4'])) {
+      if ((!empty($_REQUEST['api_key']) || !empty($_REQUEST['key']))) {
+        return (new LegacyRestAuthenticator())->auth($e, ['flow' => 'legacyrest', 'cred' => 'Bearer ' . $_REQUEST['api_key'] ?? '', 'siteKey' => $_REQUEST['key'] ?? NULL]);
```

Comment:
Haven't read/tested this branch, but I'm not sure you need to explicitly call out to `LegacyRestAuthenticator`. It should register itself.

Reply:
@totten yes I believe this was left over code from the previous version will remove